docs/geps/gep-02.rst

```diff
@@ -0,0 +1,158 @@
+.. _gep-2:
+
+=====================================================================
+GEP 2 — Internal Representation of Data on Individuals and Households
+=====================================================================
+
++------------+-------------------------------------------------------------------------+
+| Author     | `Hans-Martin von Gaudecker <https://github.com/hmgaudecker>`_           |
++------------+-------------------------------------------------------------------------+
+| Status     | Draft                                                                   |
++------------+-------------------------------------------------------------------------+
+| Type       | Standards Track                                                         |
++------------+-------------------------------------------------------------------------+
+| Created    | 2020-02-22                                                              |
++------------+-------------------------------------------------------------------------+
+| Resolution | <url> (required for Accepted | Rejected | Withdrawn)                    |
++------------+-------------------------------------------------------------------------+
+
+
+Abstract
+--------
+
+This GEP lays out how GETTSIM stores the user-provided data (be it from the SOEP, EVS,
+example indidviduals...) and passes it around to the functions calculating taxes and
+transfers.
+
+
+Motivation and Scope
+--------------------
+
+This section describes the need for the proposed change. It should describe the existing
+problem, who it affects, what it is trying to solve, and why. This section should
+explicitly address the scope of and key requirements for the proposed change.
+
+
+Usage and Impact
+----------------
+
+* Primarily internal / relevant for developers — highest-level interface can be easily
+  adjusted
+* Affects users only via the interface of lower-level functions (they might want to
+  call those or override them)
+
+
+
+Detailed description
+--------------------
+
+Main idea: adhere to normal forms (see pages https://www.wiwi.uni-bonn.de/gaudecker/_static/prog_econ/2019/09_data_management.pdf).
+
+1. Values do not have any internal structure
+2. Tables do not contain redundant information
+3. No structure in variable names (always use "long" format, not "wide" format)
+
+1. and 3. should be uncontroversial, for 2., there are two potential ways forward, see
+below.
+
+Some things that will be the case for sure:
+
+* Anything variable that exists at a level that is above the lowe
+
+.. note::
+
+    We might get by without steuersubjekt (tax unit) by just tracing out relations
+    between people and eligibility; I leave it in for now.
+
+
+A: Strict adherence to 2nd normal form
+---------------------------------------
+
+* 3+ different tables (bedarfsgemeinschaft, steuersubjekt, person, ?).
+* Indexes would be ``jahr`` :math:`\times` {``bedarfsgemeinschaft``, ``steuersubjekt``,
+  ``person``}
```

Comment:
As it turned out, bedarfsgemeinschaft and steuersubjekt are essentially the same. So `haushalt`, `bedarfsgemeinschat`, `person` might be more appropriate

Reply:
Thanks for the reminder, should have looked up [that discussion](https://gettsim.zulipchat.com/#narrow/stream/212222-general/topic/Update.20Data.20Structures/near/180917151). `person` seems clear enough. Beyond that, my understanding so far: 

- `haushalt`
   - relevant for: `wohngeld`, ...  ? 
- `steuersubjekt` / `bedarfsgemeinschaft`
    - relevant for: `e_st`, `zu_verst_e`, `kindergeld`, `arb_los_geld_2`, `kinderzuschlag`, ...? 
    - 1 or 2 persons 
    - whether kids are relevant for `zu_verst_e` / `kindergeld` or not depends on age and income, but does not depend on being part of the `haushalt`
    - whether kids are relevant for transfers like  `arb_los_geld_2` and `kinderzuschlag` depends on age, income, and being part of the `haushalt`.

Reply:
Almost 100% agree, I edited your post.

Reply:
No, I meant `haushalt`, if that is the only thing you changed. The thing is you have to live under the same roof, right?

Where this will lead to eventually, I think, is that users only specify the core adults for `steuersubjekt` / `bedarfsgemeinschaft` and we will figure out which kids to include in these things based on parent / child relationships, age, income, residence.

The problem is that these are endogenous to choices of many interesting models (labour supply of kids comes to mind), so it is essentially impossible for the user to specify them ex ante. How would you do that in a multinomial logit routine where different hours choices move a kid in or out being relevant for her parents' income tax statement?

Reply:
We'll have to check again in detail what the legal criteria are here. In IZAMOD, we don't care about kid's hours of work on the decision of whether or not they are in the `bedarfsgemeinschaft`. A kid is assinged to its parents' `bedarfsgemeinschaft` if it's below 18 or below 25 and in education. Hours of work /income matter for Kindergeld and Kinderfreibetrag, but you may have a child in the `bedarfsgemeinschaft` for which no kindergeld is paid. I don't see a major problem in allowing the `age <= 25` threshold to be flexible.

Reply:
I do not quite understand what you are trying to get at, but maybe I was not clear, either. 

I think the basic problem is that this whole stuff is endogenous and we should be correct about it rather than relying on the user to make sure that a kid who earns too much next to being a student gets to be her own `steuersubjekt`. 

And yes, we should make things like that age threshold endogenous in the sense of moving it to the params files. These have changed in the past here and there, after all.